src/test/java/fr/nelson/you_are_the_hero/service/UserServiceTest.java

```diff
@@ -0,0 +1,64 @@
+package fr.nelson.you_are_the_hero.service;
+
+import fr.nelson.you_are_the_hero.model.db.AppUser;
+import fr.nelson.you_are_the_hero.model.dto.AuthRequestDto;
+import fr.nelson.you_are_the_hero.repository.AppUserRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import org.junit.jupiter.api.Assertions;
+
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceTest {
+
+    @Mock
+    private AppUserRepository appUserRepository;
+    @Mock
+    private PasswordEncoder passwordEncoder;
+
+    @InjectMocks
+    private UserService userService;
+
+    @Test
+    public void test_createNewUser_OK() throws Exception {
+        // Given
+        AuthRequestDto authRequestDto = new AuthRequestDto();
+        authRequestDto.setPassword("my-password");
+        authRequestDto.setUsername("my-username");
+
+
+        when(appUserRepository.existsByUsername(Mockito.anyString())).thenReturn(false);
+        when(appUserRepository.save(Mockito.any())).thenReturn(new AppUser());
+        when(passwordEncoder.encode(Mockito.anyString())).thenReturn("toto");
+
+        // When
+        AppUser result = userService.saveUser(authRequestDto);
+
+        // Then
+        Assertions.assertNotNull(result);
+        verify(appUserRepository, times(1)).save(Mockito.any());
```

Comment:
the return object of the method will be : when(appUserRepository.save(Mockito.any())).thenReturn(new AppUser()); 
==> so it will be the object that I set in the mock behavior. no real utility.
and the password encoder is a third class, it can't be check in unit test and there is no integration test stack yet

Reply:
Indeed, it's a misunderstanding on my part. Thank you for your contribution.